### PR TITLE
Bugfix Octree Factory: returning the correct error if the factory function with prefix fails

### PR DESCRIPTION
**Pull request by catevita**

### CODE REVIEW

**Comment by catevita** [comment]
thanks for finding it out
PTAL @nnmm

File: src/octree/factory.rs [hunk 1]
```diff
@@ -31,9 +31,7 @@ impl OctreeFactory {
             if !octree_argument.starts_with(prefix) {
                 continue;
             }
-            if let Ok(o) = octree_factory_function(octree_argument) {
-                return Ok(o);
-            }
+           return Ok(octree_factory_function(octree_argument)?);
```
Comment (nnmm) [inline src/octree/factory.rs]: Let's only change this to remove the redundancy, as discussed. Otherwise looks good.

**Comment by catevita** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user catevita. Merge queue now has a length of 1.

**Pull request merged**